internal/oci/pull.go

```diff
@@ -117,6 +121,14 @@ func WithPullAuth(a *ImagePullAuth) ImageClientOption {
 	}
 }
 
+// WithCustomCA adds given root certificates to tls client configuration
+func WithCustomCA(rootCAs *x509.CertPool) ImageClientOption {
+	return func(c *ImageClientOptions) {
+		c.transport = remote.DefaultTransport.(*http.Transport).Clone()
```

Comment:
It's unlikely, but if `remote.DefaultTransport` stops being a `*http.Transport` in a future go-containerregistry release we'd panic at runtime. Can we add or update a unit test to call this function? I'm not worried about testing its behaviour really, I just want to make sure we invoke it at build time to make sure it doesn't panic.